p11-kit/meson.build

```diff
@@ -0,0 +1,337 @@
+              sources: 'mock-module-ep.c',
+              include_directories: [configinc, commoninc],
+              link_with: [libp11_test, libp11_common])
+
+shared_module('mock-two',
+              name_prefix: '',
+              sources: 'mock-module-ep2.c',
+              include_directories: [configinc, commoninc],
+              link_with: [libp11_test, libp11_common])
+
+shared_module('mock-three',
+              name_prefix: '',
+              sources: 'mock-module-ep.c',
+              include_directories: [configinc, commoninc],
+              link_with: [libp11_test, libp11_common])
+
+shared_module('mock-four',
+              name_prefix: '',
+              sources: 'mock-module-ep.c',
+              include_directories: [configinc, commoninc],
+              link_with: [libp11_test, libp11_common])
+
+shared_module('mock-five',
+              name_prefix: '',
+              sources: 'mock-module-ep3.c',
+              include_directories: [configinc, commoninc],
+              link_with: [libp11_test, libp11_common])
+
+if host_system != 'windows'
+  shared_module('mock-six',
+                name_prefix: '',
+                sources: 'mock-module-ep4.c',
+                include_directories: [configinc, commoninc],
+                link_with: [libp11_test, libp11_common])
+endif
+
+shared_module('mock-seven',
+              name_prefix: '',
+              sources: 'mock-module-ep5.c',
+              include_directories: [configinc, commoninc],
+              link_with: [libp11_test, libp11_common])
+
+shared_module('mock-eight',
+              name_prefix: '',
+              sources: 'mock-module-ep6.c',
+              include_directories: [configinc, commoninc],
+              link_with: [libp11_test, libp11_common])
+
+shared_module('mock-nine',
+              name_prefix: '',
+              sources: 'mock-module-ep7.c',
+              include_directories: [configinc, commoninc],
+              link_with: [libp11_test, libp11_common])
+
+pkg.generate(libraries: libp11_kit,
+             filebase: 'p11-kit-1',
+             name: 'p11-kit',
+             description: 'Library and proxy module for properly loading and sharing PKCS#11 modules.',
+             subdirs: 'p11-kit-1',
+             variables: [
+               'p11_module_configs=@0@'.format(p11_package_config_modules),
+               'p11_module_path=@0@'.format(p11_module_path),
+               # 'p11_trust_paths=@0@'.format(with_trust_paths),
+               'proxy_module=@0@/p11-kit-proxy.so'.format(libdir),
+               'p11_system_config_modules=@0@'.format(p11_package_config_modules)
+             ])
+
+pkcs11_conf_example_conf = configuration_data()
+pkcs11_conf_example_conf.set('p11_system_config_file', p11_system_config_file)
+pkcs11_conf_example_conf.set('p11_user_config', p11_user_config)
+pkcs11_conf_example = configure_file(input: 'pkcs11.conf.example.in',
+                                     output: 'pkcs11.conf.example',
```

Comment:
Alternatively `@BASENAME@` can be used so it's less error prone in name changes.